@shared/helpers/soroban/token.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  Contract,
+  TransactionBuilder,
+  Memo,
+  Server,
+  TimeoutInfinite,
+  xdr,
+} from "soroban-client";
+import { simulateTx } from "./server";
+
+export const transfer = async (
+  contractId: string,
+  params: xdr.ScVal[],
+  memo: string | undefined,
+  builder: TransactionBuilder,
+) => {
+  const contract = new Contract(contractId);
+
+  const tx = builder
+    .addOperation(contract.call("transfer", ...params))
+    .setTimeout(TimeoutInfinite);
+
+  if (memo) {
+    tx.addMemo(Memo.text(memo));
+  }
+
+  return tx.build();
+};
+
+export const getBalance = async (
+  contractId: string,
+  params: xdr.ScVal[],
+  server: Server,
+  builder: TransactionBuilder,
+) => {
+  const contract = new Contract(contractId);
+
+  const tx = builder
+    .addOperation(contract.call("balance", ...params))
+    .setTimeout(TimeoutInfinite)
+    .build();
+
+  const result = await simulateTx<number>(tx, server);
+  return result;
+};
+
+export const getDecimals = async (
+  contractId: string,
+  server: Server,
+  builder: TransactionBuilder,
+) => {
+  const contract = new Contract(contractId);
+
+  const tx = builder
+    .addOperation(contract.call("decimals"))
+    .setTimeout(TimeoutInfinite)
+    .build();
+
+  const result = await simulateTx<number>(tx, server);
+  return result;
+};
+
+export const getName = async (
+  contractId: string,
+  server: Server,
+  builder: TransactionBuilder,
+) => {
+  const contract = new Contract(contractId);
+
+  const tx = builder
+    .addOperation(contract.call("name"))
+    .setTimeout(TimeoutInfinite)
+    .build();
+
+  const result = await simulateTx<string>(tx, server);
+  return result;
+};
+
+export const getSymbol = async (
+  contractId: string,
+  server: Server,
+  builder: TransactionBuilder,
+) => {
+  const contract = new Contract(contractId);
+
+  const tx = builder
+    .addOperation(contract.call("symbol"))
+    .setTimeout(TimeoutInfinite)
+    .build();
+
+  const result = await simulateTx<string>(tx, server);
+  return result;
```

Comment:
no need to do this now since things could still change - but if this API holds, in the future we could probably refactor these helpers to share some logic as they're all kinda doing the same thing.

Also, @Shaptic - do we consider this a common enough use case (getting balance + decimals + name + symbol) that soroban-client should offer this?

Reply:
Additionally - there's also a good convo happening about how we should be fetching data for Soroban now that expired state can be involved. It seems like we'll have to at least augment our approach anyway pretty soon. Sem good ideas that came up includes -
1. Caching things like decimals/name/symbol when a user firsts gets a balance of a token.
2. Fetching the ledger entries directly(this is cheaper than the rpc call).
3. Getting these details from Horizon when token is a SAC.
4. Allowing users to restore their data when their token data is expired(implies we start storing expiration time along with other details).

Reply:
I don't think this belongs in soroban-client, per se, but maybe this be handled by using soroban-cli to generate the TypeScript bindings for the SAC? cc @chadoh @willemneal